### ISSUE TITLE
One Click Install (Linux/Netbook)

### ISSUE THREAD

**Issue by baziwane** [rapidftr/tracker#50]
Using Docker to create a one-click installer for Linux OSes. The installer will be stored in [Dropbox](https://www.dropbox.com/sh/y9cjomps39deqb6/AAAmbIS4yyvff9l5iuXPwLSua) folder available publicly.

**Comment by baziwane** [comment]
cross posting @rdsubhas notes from mailing list regarding this feature

Linux Installer for RapidFTR is up @ https://github.com/rapidftr/rapidftr-infrastructure/tree/master/installer
Run:
vagrant up linux-dev
This will boot up a Vagrant VM, install Docker, download and export RapidFTR image as a tar.gz file. It will also copy all the DEB dependencies that can be used to install Docker offline in Ubuntu 14.04.
Once the vagrant machine is run, an "artifacts" folder is created with the RapidFTR image and the DEBs. Now you can run "./install.sh" to install RapidFTR using those artifacts.
For testing purposes, I have added another Vagrant image. This image automatically has Internet disabled, so it will simulate Offline access.
Run (after generating the artifacts as said above):
vagrant up linux-test
And then inside this VM, you'll be able to run "./install.sh" and test the installer.
TL;DR - Running Vagrant will generate all the artifacts. For Windows, I'm also planning to use Vagrant 1.6's Windows support to generate the Windows installers as well.

**Comment by tomclement** [comment]
One click install tested on Ubuntu 14.04.1 LTS (Offline). Save for one error relating to `linux-image-extra-3.13.0-35-generic` install works perfectly. See full installation screen output below:

```
rapidftr@rapidftr-Intel-powered-classmate-PC:~/Desktop/installer$ sudo chmod u+x install.sh 
rapidftr@rapidftr-Intel-powered-classmate-PC:~/Desktop/installer$ sudo ./install.sh 
Docker is not installed, trying to install...
Checking if we can install Docker online...
/usr/bin/wget
DISTRIB_DESCRIPTION="Ubuntu 14.04 LTS"
Installing Docker offline...
Selecting previously unselected package aufs-tools.
(Reading database ... 163326 files and directories currently installed.)
Preparing to unpack .../aufs-tools_1%3a3.2+20130722-1.1_amd64.deb ...
Unpacking aufs-tools (1:3.2+20130722-1.1) ...
Selecting previously unselected package cgroup-lite.
Preparing to unpack .../apt/cgroup-lite_1.9_all.deb ...
Unpacking cgroup-lite (1.9) ...
Preparing to unpack .../crda_1.1.2-1ubuntu2_amd64.deb ...
Unpacking crda (1.1.2-1ubuntu2) over (1.1.2-1ubuntu2) ...
Selecting previously unselected package git.
Preparing to unpack .../apt/git_1%3a1.9.1-1_amd64.deb ...
Unpacking git (1:1.9.1-1) ...
Selecting previously unselected package git-man.
Preparing to unpack .../git-man_1%3a1.9.1-1_all.deb ...
Unpacking git-man (1:1.9.1-1) ...
Preparing to unpack .../apt/iw_3.4-1_amd64.deb ...
Unpacking iw (3.4-1) over (3.4-1) ...
Selecting previously unselected package liberror-perl.
Preparing to unpack .../liberror-perl_0.17-1.1_all.deb ...
Unpacking liberror-perl (0.17-1.1) ...
Preparing to unpack .../libnl-3-200_3.2.21-1_amd64.deb ...
Unpacking libnl-3-200:amd64 (3.2.21-1) over (3.2.21-1) ...
Preparing to unpack .../libnl-genl-3-200_3.2.21-1_amd64.deb ...
Unpacking libnl-genl-3-200:amd64 (3.2.21-1) over (3.2.21-1) ...
Selecting previously unselected package linux-image-extra-3.13.0-35-generic.
Preparing to unpack .../linux-image-extra-3.13.0-35-generic_3.13.0-35.62_amd64.deb ...
Unpacking linux-image-extra-3.13.0-35-generic (3.13.0-35.62) ...
Selecting previously unselected package lxc-docker-1.2.0.
Preparing to unpack .../lxc-docker-1.2.0_1.2.0_amd64.deb ...
Unpacking lxc-docker-1.2.0 (1.2.0) ...
Selecting previously unselected package lxc-docker.
Preparing to unpack .../apt/lxc-docker_1.2.0_amd64.deb ...
Unpacking lxc-docker (1.2.0) ...
Preparing to unpack .../wireless-regdb_2013.02.13-1ubuntu1_all.deb ...
Unpacking wireless-regdb (2013.02.13-1ubuntu1) over (2013.02.13-1ubuntu1) ...
Setting up aufs-tools (1:3.2+20130722-1.1) ...
Setting up cgroup-lite (1.9) ...
cgroup-lite start/running
Setting up git-man (1:1.9.1-1) ...
Setting up liberror-perl (0.17-1.1) ...
Setting up libnl-3-200:amd64 (3.2.21-1) ...
Setting up libnl-genl-3-200:amd64 (3.2.21-1) ...
dpkg: dependency problems prevent configuration of linux-image-extra-3.13.0-35-generic:
 linux-image-extra-3.13.0-35-generic depends on linux-image-3.13.0-35-generic; however:
  Package linux-image-3.13.0-35-generic is not installed.
 
dpkg: error processing package linux-image-extra-3.13.0-35-generic (--install):
 dependency problems - leaving unconfigured
Setting up lxc-docker-1.2.0 (1.2.0) ...
docker start/running, process 3196
Setting up wireless-regdb (2013.02.13-1ubuntu1) ...
Setting up crda (1.1.2-1ubuntu2) ...
Setting up git (1:1.9.1-1) ...
Setting up iw (3.4-1) ...
Processing triggers for ureadahead (0.100.0-16) ...
ureadahead will be reprofiled on next reboot
Setting up lxc-docker (1.2.0) ...
Processing triggers for man-db (2.6.7.1-1) ...
Processing triggers for libc-bin (2.19-0ubuntu6) ...
Errors were encountered while processing:
 linux-image-extra-3.13.0-35-generic
Removing old RapidFTR...
Installing RapidFTR...
b4dbcc4d8271c305faef08a45636c77ed6c8f55e98ddfacbcc0ea8607a5682a7
3e0810e084d1402f12e248fbcf13be95d7e95b0994624c426b13a733b05f16d6
Cleaning up...
```

**Comment by tomclement** [comment]
The documentation for this story has been added at:
- http://rapidftr.github.io/guide/deploy/web/generating-installers.html
- http://rapidftr.github.io/guide/deploy/web/linux-netbook.html

**Comment by tomclement** [comment]
Blocking this due to https://github.com/rapidftr/tracker/issues/227

**Comment by tomclement** [comment]
Closing this as #227 was resolved by https://github.com/rapidftr/rapidftr-infrastructure/commit/1ea5ea30512314975bb4144d49a1b408c8b5a5ce
